ShaTuApp/src/main/java/edu/regis/shatu/view/StepView.java

```diff
@@ -1,237 +1,219 @@
-        add(shaOne, StepSelection.SHA_ONE.toString());
-        add(add1View, StepSelection.ADD1.toString());
-        add(padView, StepSelection.PAD.toString());
-        add(rotateView, StepSelection.ROTATE_BITS.toString());
-        add(initVarView, StepSelection.INIT_VARS.toString());
-        add(exclusiveOrView, StepSelection.XOR.toString());
-        add(choiceFunctionView, StepSelection.CHOICE_FUNCTION.toString());
-        
-        add(stepReplyView, StepSelection.STEP_REPLY.toString());
-    }
-}
+/*
+ * SHATU: SHA-256 Tutor
+ * 
+ *  (C) Johanna & Richard Blumenthal, All rights reserved
+ * 
+ *  Unauthorized use, duplication or distribution without the authors'
+ *  permission is strictly prohibted.
+ * 
+ *  Unless required by applicable law or agreed to in writing, this
+ *  software is distributed on an "AS IS" basis without warranties
+ *  or conditions of any kind, either expressed or implied.
+ */
+package edu.regis.shatu.view;
+
+import edu.regis.shatu.err.IllegalArgException;
+import edu.regis.shatu.model.TutoringSession;
+import java.awt.CardLayout;
+import java.awt.Color;
+import javax.swing.BorderFactory;
+import javax.swing.JPanel;
+
+/**
+ * A view displaying a current task and step in the tutoring session, which
+ * typically allows a student to practice a specific aspect of the SHA-256
+ * algorithm, such as performing the XOR function.
+ * 
+ * @author rickb
+ */
+public class StepView extends JPanel {       
+    /**
+     * The name of the card (child view) currently displayed in this view.
+     * Pass this value to getUserRequestView(...) to obtain the actual view.
+     */
+    private StepSelection selectedPanel;
+    
+    /**
+     * The child ASCII child view (card) that can be displayed in this view 
+     */
+    private EncodeView encodeView;
+    
+    private CompressionCanvasView compressionView;
+    
+    private PrepareScheduleView prepareScheduleView;
+    
+    private Add1View add1View;
+    
+    private PadView padView;
+    
+    private RotateView rotateView;
+    
+    private InitVarView initVarView;
+
+    private ShiftRightView shiftRightView;
+    
+    private ExclusiveOrView exclusiveOrView;
+
+    private AddTwoBitView addTwoBitView;
+    
+    private MajFunctionView majFunction;
+    
+    private ShaZeroView shaZero;
+    
+    private ShaOneView shaOne;
+    
+    private ChoiceFunctionView choiceFunctionView;
+    
+    private StepCompletionReplyView stepReplyView;
+    
+    private TutoringSession model;
+   
+    /**
+     * Initialize and layout the child components (cards) displayed in this view.
+     */
+    public StepView() {
+        GuiController.instance().setStepView(this);
+        
+        setLayout(new CardLayout());
+        
+        initializeComponents();
+        initializeLayout();
+        
+        setBorder(BorderFactory.createLineBorder(Color.BLACK));
+        selectPanel(StepSelection.ENCODE);   
```

Comment:
Another public method named selectPanel exists in SplashFrame.java. Maybe reference this method like:
this.selectPanel(StepSelection.Encode) to ensure students know the selectPanel method from this class is being called? I do not think this change is needed, but it may help future students when interpreting this file.